scripts/build-container-image.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# This script is meant to be used by CI, to build and push container images to the container hub.
+# However, it can be used by human on a linux environment, with the following requirements:
+#  - An existing credentials towards the hub, with push and delete permissions.
+#  - Needed tools (git, docker, curl, etc. are installed)
+#
+HUB="quay.io"
+REPO="metal3-io"
+IMAGE_NAME=$1
+BRANCH_NAME=${2:-main}
+KEEP_TAGS=${3:-3}
+
+__dir__=$(realpath "$(dirname "$0")")
+IMAGES_JSON="$__dir__/files/container_image_names.json"
+
+function list_tags() {
+  curl -s "https://${HUB}/v2/${REPO}/${IMAGE_NAME}/tags/list" | jq -r '.tags[]'
+}
+
+function get_tag_sha256() {
+  tag=$1
+  SHA_REQ=$(curl -s -H "Accept: application/vnd.docker.distribution.manifest.v2+json" "https://${HUB}/v2/${REPO}/${IMAGE_NAME}/manifests/${tag}" | jq -r '.config.digest')
+  SHA=$(echo "$SHA_REQ" | cut -f 2- -d ":" | tr -d '[:space:]')
+  echo "$SHA"
+}
+
+function delete_tag() {
+  # Untested
+  tag=$1
+  sha256=$(get_tag_sha256 "$tag")
+  curl -X DELETE -s "https://${HUB}/v2/${REPO}/${IMAGE_NAME}/manifests/${sha256}"
+}
+
+function git_get_current_commit_short_hash() {
+  git rev-parse --short HEAD
+}
+
+function git_get_current_branch() {
+  git rev-parse --abbrev-ref HEAD
+}
+
+function get_date() {
+  # Should this be current date, or latest commit date?
+  date +%Y%m%d
+}
+
+function get_image_tag() {
+  echo "$(git_get_current_branch)_$(get_date)_$(git_get_current_commit_short_hash)"
+}
+
+function cleanup_old_tags() {
+  all_tags=$(list_tags)
+  branch=$(git_get_current_branch)
+  branch_tags=()
```

Comment:
```suggestion
  declare -a branch_tags=()
```
Arrays are better off declared as such.